### ISSUE TITLE
Nunit30Settings.xml becomes corrupted

### ISSUE THREAD

**Issue by alexv0** [nunit/nunit#1061]
While trying to update a project with some failing tests (using NUnitForms), in some cases Nunit30Settings.xml becomes corrupted and is filled with a bunch of "nul"s, which, on the next run of nunit3-console results in the following exception:
```
     [exec] System.ApplicationException: Error loading settings file ---> System.Xml.XmlException: Root element is missing.
     [exec]    at System.Xml.XmlTextReaderImpl.ThrowWithoutLineInfo(String res)
     [exec]    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
     [exec]    at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean reserveWhitespace)
     [exec]    at System.Xml.XmlDocument.Load(XmlReader reader)
     [exec]    at System.Xml.XmlDocument.Load(String filename)
     [exec]    at NUnit.Engine.Internal.SettingsStore.LoadSettings()
     [exec]    --- End of inner exception stack trace ---
     [exec]    at NUnit.Engine.Internal.SettingsStore.LoadSettings()
     [exec]    at NUnit.Engine.Services.SettingsService.StartService()
     [exec]    at NUnit.Engine.Services.ServiceManager.StartServices()
     [exec]    at NUnit.Engine.TestEngine.Initialize()
     [exec]    at NUnit.Engine.TestEngine.NUnit.Engine.ITestEngine.get_Services()
     [exec]    at NUnit.ConsoleRunner.ConsoleRunner..ctor(ITestEngine engine, ConsoleOptions options, ExtendedTextWriter writer)
     [exec]    at NUnit.ConsoleRunner.Program.Main(String[] args)
```

As a workaround, one can delete this file, but it is pretty annoying and most importantly it wasn't clear what settings file it was referring to in the first place.

**Comment by CharliePoole** [comment]
We can improve the message to include the name of the file. I don't see that there is anything we can do about the corruption if it's caused by non-NUnit software. Did you have any suggestions?

**Comment by CharliePoole** [comment]
To clarify my last remark: NUnitForms an addon that works with NUnit. It isn't developed by the NUnit team, although my own name is listed as one of the developers. That's because we once planned to integrate it with NUnit, but we lost momentum and it never happened.

As far as I know, it has not been updated in any way to work with NUnit 3.0. It may work, or mostly work, because the coupling is pretty loose. I'll be glad to work with anyone who is trying to update it to work well with NUnit 3.0.

**Comment by alexv0** [comment]
I think including the name of the file should do it.  Thanks.

Yup, I'm aware that NUnitForms is a project separate from NUnit and it does look like it has been abandoned for quite some time.  We use it due to the fact that the base functionality still mostly works and I haven't been able to find a better alternative.  I'll let you know how the update to NUnit3 goes.

**Comment by CharliePoole** [comment]
@alexv0 Let me know how it goes with NUnitForms. Since I am a commiter, it's possible to make some changes to it if necessary for NUnit 3.0.

**Comment by rprouse** [comment]
Pull request for this is pending. I just changed it to include the filename in the message. It now gives the following output.

```
System.ApplicationException: Error loading settings C:\Users\rob.prouse\AppData\Local\NUnit\Nunit30Settings.xml. ---> System.Xml.XmlException: The 'Settings' start tag on line 3 position 4 does not match the end tag of 'NUnitSettings'. Line 4, position 3.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ThrowTagMismatch(NodeData startTag)
   at System.Xml.XmlTextReaderImpl.ParseEndElement()
   at System.Xml.XmlTextReaderImpl.ParseElementContent()
   at System.Xml.XmlLoader.LoadNode(Boolean skipOverWhitespace)
   at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)
   at System.Xml.XmlDocument.Load(XmlReader reader)
   at System.Xml.XmlDocument.Load(String filename)
   at NUnit.Engine.Internal.SettingsStore.LoadSettings() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\Internal\SettingsStore.cs:line 74
   --- End of inner exception stack trace ---
   at NUnit.Engine.Internal.SettingsStore.LoadSettings() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\Internal\SettingsStore.cs:line 93
   at NUnit.Engine.Services.SettingsService.StartService() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\Services\SettingsService.cs:line 57
   at NUnit.Engine.Services.ServiceManager.StartServices() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\Services\ServiceManager.cs:line 95
   at NUnit.Engine.TestEngine.Initialize() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\TestEngine.cs:line 110
   at NUnit.Engine.TestEngine.NUnit.Engine.ITestEngine.get_Services() in D:\Src\nunit\nunit\src\NUnitEngine\nunit.engine\TestEngine.cs:line 68
   at NUnit.ConsoleRunner.ConsoleRunner..ctor(ITestEngine engine, ConsoleOptions options, ExtendedTextWriter writer) in D:\Src\nunit\nunit\src\NUnitConsole\nunit3-console\ConsoleRunner.cs:line 81
   at NUnit.ConsoleRunner.Program.Main(String[] args) in D:\Src\nunit\nunit\src\NUnitConsole\nunit3-console\Program.cs:line 121
```

**Comment by rprouse** [comment]
Closed with PR #1076
